### PR TITLE
refact(client): update tests for new OLTP traverser APIs

### PR DESCRIPTION
**Pull request by DanGuge**
## Purpose of the PR

- related PR #542   

## Main Changes

Updata api tests with new traverser structure

## Verifying these changes

- [ ] Trivial rework / code cleanup without any test coverage. (No Need)
- [ ] Already covered by existing tests, such as *(please modify tests here)*.
- [ ] Need tests and can be verified as follows:
    - xxx

## Does this PR potentially affect the following parts?

- [ ]  Nope
- [ ]  Dependencies (add/update license info) 
- [ ]  Modify configurations
- [ ]  The public API
- [x]  Other affects (typed here)

## Documentation Status

- [ ]  `Doc - TODO` 
- [ ]  `Doc - Done` 
- [x]  `Doc - No Need`

### CODE REVIEW

File: .github/workflows/client-ci.yml [hunk 1]
```diff
@@ -24,8 +24,7 @@ jobs:
     env:
       USE_STAGE: 'true' # Whether to include the stage repository.
       TRAVIS_DIR: hugegraph-client/assembly/travis
-      # TODO: replace it with the (latest - n) commit id (n >= 15)
-      COMMIT_ID: be6ee386b9939dc6bd6fcbdf2274b8acc3a0a314
+      COMMIT_ID: 47aa8be8508293bbda76c93b461292efc84a75c7
```
Comment (simon824) [inline .github/workflows/client-ci.yml:27]: update to bfe9fae150446857412db23ada0dae9d05035837

File: .github/workflows/hubble-ci.yml [hunk 2]
```diff
@@ -23,8 +23,8 @@ on:
 
 env:
   TRAVIS_DIR: hugegraph-hubble/hubble-dist/assembly/travis
-  # TODO: replace it with the (latest - n) commit id (n >= 15)
-  COMMIT_ID: be6ee386b9939dc6bd6fcbdf2274b8acc3a0a314
+  # TODO: need update it later (eed6103359fe40d2f1476fb8c56d9388c3111a99)
+  COMMIT_ID: ed493f3093eab293a75cd2f539e29ca11fa6cd81
```
Comment (simon824) [inline .github/workflows/hubble-ci.yml:27]: ditto

File: .github/workflows/loader-ci.yml [hunk 3]
```diff
@@ -26,8 +26,7 @@ jobs:
       USE_STAGE: 'true' # Whether to include the stage repository.
       TRAVIS_DIR: hugegraph-loader/assembly/travis
       STATIC_DIR: hugegraph-loader/assembly/static
-      # TODO: replace it with the (latest - n) commit id (n >= 15)
-      COMMIT_ID: be6ee386b9939dc6bd6fcbdf2274b8acc3a0a314
+      COMMIT_ID: ed493f3093eab293a75cd2f539e29ca11fa6cd81
```
Comment (simon824) [inline .github/workflows/loader-ci.yml:29]: ditto

File: .github/workflows/tools-ci.yml [hunk 4]
```diff
@@ -25,7 +25,7 @@ jobs:
       USE_STAGE: 'true' # Whether to include the stage repository.
       TRAVIS_DIR: hugegraph-tools/assembly/travis
       # TODO: could we use one param to unify it? or use a action template (could use one ci file)
-      COMMIT_ID: be6ee386b9939dc6bd6fcbdf2274b8acc3a0a314
+      COMMIT_ID: ed493f3093eab293a75cd2f539e29ca11fa6cd81
```
Comment (simon824) [inline .github/workflows/tools-ci.yml:28]: ditto

**Comment by simon824** [comment]
ci error
```
2023-12-09T14:30:28.1358860Z 	at org.apache.hugegraph.functional.MetricsManagerTest.testAllMetrics(MetricsManagerTest.java:49)
2023-12-09T14:30:28.1359093Z Caused by: com.fasterxml.jackson.core.JsonParseException: 
2023-12-09T14:30:28.1359682Z Unexpected character ('#' (code 35)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
2023-12-09T14:30:28.1359848Z  at [Source: (StringReader); line: 1, column: 2]
2023-12-09T14:30:28.1360258Z 	at org.apache.hugegraph.functional.MetricsManagerTest.testAllMetrics(MetricsManagerTest.java:49)
```

**Comment by DanGuge** [comment]
> ci error
> 
> ```
> 2023-12-09T14:30:28.1358860Z 	at org.apache.hugegraph.functional.MetricsManagerTest.testAllMetrics(MetricsManagerTest.java:49)
> 2023-12-09T14:30:28.1359093Z Caused by: com.fasterxml.jackson.core.JsonParseException: 
> 2023-12-09T14:30:28.1359682Z Unexpected character ('#' (code 35)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
> 2023-12-09T14:30:28.1359848Z  at [Source: (StringReader); line: 1, column: 2]
> 2023-12-09T14:30:28.1360258Z 	at org.apache.hugegraph.functional.MetricsManagerTest.testAllMetrics(MetricsManagerTest.java:49)
> ```

due to changes in [MetricsAPI.java](https://github.com/apache/incubator-hugegraph/commits/bce6d058f5d1f108b4c19e18eb7a65ab0604a526/hugegraph-api/src/main/java/org/apache/hugegraph/api/metrics/MetricsAPI.java?browsing_rename_history=true&new_path=hugegraph-server/hugegraph-api/src/main/java/org/apache/hugegraph/api/metrics/MetricsAPI.java&original_branch=master)

cc @imbajin

File: hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VEStepEntity.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        return new Builder();
+    }
+
+    @Override
+    public String toString() {
+        return String.format("VEStepEntity{label=%s,properties=%s}",
+                             this.label, this.properties);
+    }
+
+    public static class Builder {
```
Comment (javeme) [inline hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VEStepEntity.java:47]: ditto

File: hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VESteps.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    @Override
+    public String toString() {
+        return String.format("Steps{direction=%s,maxDegree=%s," +
+                             "skipDegree=%s,vSteps=%s,eSteps=%s}",
+                             this.direction, this.maxDegree,
+                             this.skipDegree, this.vSteps, this.eSteps);
+    }
+
+    public static class Builder {
```
Comment (javeme) [inline hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VESteps.java:64]: ditto

File: hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VEStepEntity.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+ */
+
+package org.apache.hugegraph.structure.traverser;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class VEStepEntity {
```
Comment (javeme) [inline hugegraph-client/src/main/java/org/apache/hugegraph/structure/traverser/VEStepEntity.java:25]: rename to VEStep?

File: hugegraph-client/src/test/java/org/apache/hugegraph/api/auth/TokenApiTest.java [hunk 8]
```diff
@@ -58,6 +59,7 @@ public static void clear() {
     }
 
     @Test
+    @Ignore
```
Comment (javeme) [inline hugegraph-client/src/test/java/org/apache/hugegraph/api/auth/TokenApiTest.java:62]: revert it?

File: hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/CountApiTest.java [hunk 9]
```diff
@@ -253,6 +254,7 @@ public void testCountWithLabel() {
     }
 
     @Test
+    @Ignore
```
Comment (javeme) [inline hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/CountApiTest.java:257]: ditto

File: hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java [hunk 10]
```diff
@@ -372,9 +390,14 @@ public void testKneighborPostWithProperties() {
 
         builder = KneighborRequest.builder();
         builder.source(markoId);
-        builder.step().direction(Direction.BOTH)
-               .properties("date", "P.gt(\"2014-01-01 00:00:00\")");
+        builder.steps()
+               .direction(Direction.BOTH)
```
Comment (javeme) [inline hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java:394]: we can move direction into VEStepEntity in the future

File: hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java [hunk 11]
```diff
@@ -385,9 +408,15 @@ public void testKneighborPostWithProperties() {
...
+               .eSteps(VEStepEntity
```
Comment (javeme) [inline hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/KneighborApiTest.java:413]: can we simplify to this style: `builder.addVertexStep(...).addEdgeStep(...).addEdgeStep(...)`, and we can provide multi versions like
```java
addEdgeStep(label);
addEdgeStep(property1-name, property1-value, property2-name, property2-value, ...);
addEdgeStep(label, property1-name, property1-value, property2-name, property2-value, ...);
```

File: hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/CountApiTest.java [hunk 12]
```diff
@@ -253,6 +254,7 @@ public void testCountWithLabel() {
...
     public void testCountWithProperties() {
```
Comment (imbajin) [inline hugegraph-client/src/test/java/org/apache/hugegraph/api/traverser/CountApiTest.java:258]: Note：This test ignore now & need handle it later

**Review by imbajin — APPROVED** [review]

**Review by simon824 — APPROVED** [review]

**Pull request merged**